main/src/ca/uwaterloo/flix/language/phase/jvm/GenRecordExtend.scala

```diff
@@ -0,0 +1,227 @@
+    * The `equals` method is always the following:
+    *
+    * public boolean equals(Object var1) throws Exception {
+    * throw new Exception("equals method shouldn't be called");
+    * }
+    *
+    */
+  private def genByteCode(classType: JvmType.Reference, interfaceType: JvmType.Reference, targs: List[JvmType])(implicit root: Root, flix: Flix): Array[Byte] = {
+    // class writer
+    val visitor = AsmOps.mkClassWriter()
+
+    // internal name of super
+    val superClass = JvmName.Object.toInternalName
+
+    // internal name of interfaces to be implemented
+    val implementedInterfaces = Array(interfaceType.name.toInternalName)
+
+    // Initialize the visitor to create a class.
+    visitor.visit(AsmOps.JavaVersion, ACC_PUBLIC + ACC_FINAL, classType.name.toInternalName, null, superClass, implementedInterfaces)
+
+    // Source of the class
+    visitor.visitSource(classType.name.toInternalName, null)
+
+    for ((field, ind) <- targs.zipWithIndex) {
+      // Name of the field
+      val fieldName = s"field$ind"
+
+      // Defining fields of the tuple
+      AsmOps.compileField(visitor, fieldName, field, isStatic = false, isPrivate = true)
+    }
+
+    // Emit the code for the constructor
+    compileRecordExtendConstructor(visitor, classType, targs)
+
+    // Emit code for the 'getField' method
+    compileRecordExtendGetField(visitor, classType)
+
+    // Generate `toString` method
+    AsmOps.compileExceptionThrowerMethod(visitor, ACC_PUBLIC + ACC_FINAL, "toString", AsmOps.getMethodDescriptor(Nil, JvmType.String),
+      "toString method shouldn't be called")
+
+    // Generate `hashCode` method
+    AsmOps.compileExceptionThrowerMethod(visitor, ACC_PUBLIC + ACC_FINAL, "hashCode", AsmOps.getMethodDescriptor(Nil, JvmType.PrimInt),
+      "hashCode method shouldn't be called")
+
+    // Generate `equals` method
+    AsmOps.compileExceptionThrowerMethod(visitor, ACC_PUBLIC + ACC_FINAL, "equals", AsmOps.getMethodDescriptor(List(JvmType.Object), JvmType.Void),
+      "equals method shouldn't be called")
+
+    visitor.visitEnd()
+    visitor.toByteArray
+  }
+
+
+  /**
+    * This method generates the constructor for the RecordEmpty class. This constructor doesn't receive any arguments.
+    * For example for RecordEmpty() creates the following constructor:
+    *
+    * public RecordEmpty() {}
+    */
+  def compileRecordExtendConstructor(visitor: ClassWriter, classType: JvmType.Reference, fields: List[JvmType])(implicit root: Root, flix: Flix): Unit = {
+
+    val constructor = visitor.visitMethod(ACC_PUBLIC, "<init>", AsmOps.getMethodDescriptor(fields, JvmType.Void), null, null)
+
+    constructor.visitCode()
+    constructor.visitVarInsn(ALOAD, 0)
+
+    // Call the super (java.lang.Object) constructor
+    constructor.visitMethodInsn(INVOKESPECIAL, JvmName.Object.toInternalName, "<init>", AsmOps.getMethodDescriptor(Nil, JvmType.Void), false)
+
+
```

Comment:
Yuck... but that is why we are trying to come up with a cleaner framework for the backend :)